app/Http/Controllers/DashboardController.php

```diff
@@ -5,17 +5,52 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\DB;
 use Illuminate\Support\Facades\Auth;
+use App\Models\UserFinanceAffiliation;
+use App\Models\CustomerFinanceOrganizationAffiliation;
 
 class DashboardController extends Controller
-{
+{   
+    public function __construct() {
+        $this->middleware('auth');
+    }
+
     public function index() {
-         $order = DB::table('orders')
-        ->leftJoin('status_options', 'orders.status_id', '=', 'status_options.id')
-        ->join('order_details', 'order_details.order_id', '=', 'orders.id')
-        ->join('products','order_details.product_id', '=', 'products.id')
-        ->select('orders.*','status_options.status_name','products.image_one_secure_url','order_details.product_id')
-        ->where('orders.user_id',Auth::id())->orderBy('orders.id','DESC')->limit(10)->get();
-
-        return view('dashboard',compact('order'));
+        $lender_organizations = CustomerFinanceOrganizationAffiliation::where('user_id', Auth::id())->get();
+        $selected_organizations = UserFinanceAffiliation::where('user_id', Auth::id())->get();
+        
+        if($lender_organizations->isNotEmpty() || $selected_organizations->isNotEmpty()) {
+            
+              $order = DB::table('orders')
+              ->leftJoin('status_options', 'orders.status_id', '=', 'status_options.id')
+              ->join('order_details', 'order_details.order_id', '=', 'orders.id')
+              ->join('products','order_details.product_id', '=', 'products.id')
+              ->select('orders.*','status_options.status_name','products.image_one_secure_url','order_details.product_id')
+              ->where('orders.user_id',Auth::id())->orderBy('orders.id','DESC')->limit(10)->get();
+
+              return view('dashboard',compact('order'));
+        }else{ 
+            $finance_institutions = DB::table('lenders')->get();
+            return view('pages.select_finance_institution', compact('finance_institutions'));
+        }  
+    }
+
+     public function saveFinanceInstitution(Request $request) {
+     $finance_institution = $request->selectedInstitutions;
+     $identification_card = $request->input('identification');
+     $identification_number = $request->input('identification_number');
+
+     $data = array();
+     for ($x = 0; $x < sizeof($finance_institution); $x++) { $arr=array(); $arr['user_id']=Auth::user()->id;
+         $arr['lender_organization_id'] = $finance_institution[$x];
+         $arr['identification_type'] = $identification_card;
+         $arr['identification_number'] = $identification_number;
+         $arr['created_at'] = now();
+         array_push($data, $arr);
+         }
+
+    UserFinanceAffiliation::insert($data);
+
+    $insertedId = DB::getPdo()->lastInsertId();
```

Comment:
How to properly get the last inserted id when using models clases
https://www.larashout.com/laravel-8-get-last-id-of-an-inserted-model

Reply:
Because of inserting multiple arrays, the last method specified in the link you sent me which is the same as the one I have currently seems to be the fix for getting the last inserted id currently. I have tried the other methods but they aren't working currently.